### PR TITLE
feat: adds logging backend method

### PR DESCRIPTION
**Pull request by thisislawatts**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Developer experience (improves developer workflows for contributing to the project)

## Description

Introduces a serverless function which writes data to Google's Cloud Logging platform: 

```bash
curl -X POST https://REGION-PROJECT_ID.cloudfunctions.net/logToCloudLogging-logToCloudLogging -H "Content-Type:application/json" -d '{"message":"Hello, World!"}'
```

This could then be used with by the client side logger introduced as part of #2265 

Outstanding work:
* [x] Introduce tests that verify functional correctness
* [x] Handle malformed requests

### CODE REVIEW

File: functions/src/logging/logging.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import * as functions from 'firebase-functions'
+import { Logging } from '@google-cloud/logging'
+import { json } from 'body-parser'
+const jsonParser = json({
+  strict: true,
+})
+
+const logger = new Logging()
+
+export const handlelogToCloudLogging =
```
Comment (evakill) [inline functions/src/logging/logging.ts:10]: nit: camel case `handleLogToCloudLogging`

File: functions/src/logging/logging.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      // If not, return a 400 error status code.
+      if (typeof req.body !== 'object' || req.body === null) {
+        return res.status(400).send('Request body must be a JSON object')
+      }
+
+      if (!Object.keys(req.body).length) {
+        return res.status(400).send('Request body must not be an empty object')
+      }
+
+      // Perform some validation on the request body
```
Comment (evakill) [inline functions/src/logging/logging.ts:41]: comment on 41 seems like it refers to the checks on 31-39 - remove or move up?

File: functions/src/logging/logging.ts (hunk 1 above)
Comment (thisislawatts) [reply functions/src/logging/logging.ts:10]: Updated

File: functions/src/logging/logging.ts (hunk 2 above)
Comment (thisislawatts) [reply functions/src/logging/logging.ts:41]: Ah this is a note about putting some kind verification in that endpoint only logs payloads which match an expected schema. Will remove for the time being :)

**Review by evakill — APPROVED** [review]
LGTM

**Pull request merged**